rotkehlchen/chain/evm/decoding/metamask/decoder.py

```diff
@@ -0,0 +1,177 @@
+                out_event = event
+
+        # maybe reshuffle out first then in
+        maybe_reshuffle_events(
+            ordered_events=[out_event, in_event],
+            events_list=context.decoded_events,
+        )
+
+        # extract the fee info
+        fee_raw = fee_asset = None
+        for log in context.all_logs:
+            if log.topics[0] == METAMASK_FEE_TOPIC:
+                fee_raw = hex_or_bytes_to_int(log.data)
+                fee_asset = self.native_currency
+                break
+            if (
+                log.topics[0] == ERC20_OR_ERC721_TRANSFER and
+                log.topics[2] == self.gnosis_safe_proxy
+            ):
+                fee_raw = hex_or_bytes_to_int(log.data)
+                if log.address in in_event.asset.identifier:  # type: ignore
+                    fee_asset = in_event.asset  # type: ignore
+                else:
+                    fee_asset = out_event.asset  # type: ignore
+                break
+
+        if not (fee_raw and fee_asset):
+            return DEFAULT_DECODING_OUTPUT
+
+        fee_asset_symbol = fee_asset.resolve_to_asset_with_symbol().symbol
+        fee_amount = asset_normalized_value(amount=fee_raw, asset=fee_asset)  # type: ignore
+
+        for event in context.decoded_events:
+            # Edit the full amount in the swap's receive event
+            if (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.RECEIVE and
+                event.counterparty == CPT_METAMASK and
+                event.asset == fee_asset
+            ):
+                full_amount = to_amount + fee_amount
+                event.balance.amount = full_amount
+                event.notes = f'Receive {full_amount} {fee_asset_symbol} from {CPT_METAMASK} swap in {event.location_label}'  # noqa: E501
+                break
+
+        # And now create a new event for the fee
+        fee_event = self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=fee_asset,
+            balance=Balance(amount=fee_amount),
+            location_label=sender,
+            notes=f'Deduct {fee_amount} {fee_asset_symbol} from {sender} as {CPT_METAMASK} fees',
+            counterparty=CPT_METAMASK,
+            address=context.transaction.to_address,
+        )
+
+        return DecodingOutput(event=fee_event)
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            self.router_address: (self._decode_swap,),
+        }
+
+    @staticmethod
+    def counterparties() -> tuple[CounterpartyDetails, ...]:
+        return (CounterpartyDetails(
+            identifier=CPT_METAMASK,
+            label=METAMASK_LABEL,
+            image=METAMASK_ICON,
```

Comment:
If METAMASK_LABEL and METAMASK_ICON are only used here you can put them directly insted of constants.